pkg/auth/requests/authenticate.go

```diff
@@ -176,11 +174,22 @@ func (a *tokenAuthenticator) Authenticate(req *http.Request) (*AuthenticatorResp
 		a.refreshUser(token.UserID, false)
 	}
 
-	authResp.IsAuthed = true
-	authResp.User = token.UserID
-	authResp.UserPrincipal = token.UserPrincipal.Name
-	authResp.Groups = groups
-	authResp.Extras = getUserExtraInfo(token, authUser, attribs)
+	extras := map[string][]string{
+		common.ExtraRequestTokenID: {token.Name},
+		common.ExtraRequestHost:    {req.Host},
+	}
+	for key, value := range getUserExtraInfo(token, authUser, attribs) {
+		extras[key] = value
+	}
+
+	authResp := &AuthenticatorResponse{
+		IsAuthed:      true,
+		User:          token.UserID,
+		UserPrincipal: token.UserPrincipal.Name,
```

Comment:
One of the pieces of info needed by the token store is the user's principal ID.
Is this the information we seek ? If yes, please place it into extras well.
Because in the caller `ToAuthMiddleWare` this information is not passed on.

And further, as we have the full token here, please place the `authProvider` into the `Extras` as well.

That way the ext token store has the information it needs, __without having to refetch__ the token nor having to deal with ext vs norman tokens. That I know to add in one of the commits where ext token support is put into the auth chain. The store doesn't have to know about the type of the token at all.

Reply:
Proposed keys `ExtraRequestTokenPrincipalID`, and `ExtraRequestTokenAuthProvider`

Reply:
That's what I had in my initial POC for exactly that reason to avoid refetching the token. An issue is that a `Principal` is more than just the principal id, it also has a type, an auth provider and other attributes. The question becomes are we going to be adding all that as separate extras if/when we may find ourselves in a situation where we need to check use them without guessing/inferring their values?
I'm not saying we can't do that but it's something that needs to be considered.

Reply:
Direct discussion with @pmatseykanets ... Keeping to refetch in the token store now.